### PR TITLE
Add ilps22qs sensor support to lps2xdf driver

### PR DESCRIPTION
**Pull request by avisconti**
The ILPS22QS is an ultra-compact piezoresistive absolute pressure and temperature sensor from ST Microelectronics.
https://www.st.com/en/mems-and-sensors/ilps22qs.html

It is register compatible with lps2xdf, which already handles lps22df and lps28dfw, so we will use it to handle ilps22qs as well. During the implementation few issues/improvements have been found and fixed.

The new driver has been verified on STM steval_stwinbx1 board.

### CODE REVIEW

File: tests/drivers/build_all/sensor/i2c.dtsi [hunk 1]
```diff
@@ -1087,3 +1087,12 @@ test_i2c_tmp1075: tmp1075@98 {
...
 	interrupt-mode;
 };
+
+test_i2c_ilps22qs: ilps22qs@99 {
+	compatible = "st,ilps22qs";
+	reg = <0x99>;
+	status = "okay";
+	odr = <LPS2xDF_DT_ODR_10HZ>;
+	lpf = <LPS2xDF_DT_LP_FILTER_ODR_4>;
+	avg = <LPS2xDF_DT_AVG_128_SAMPLES>;
```
Comment (jonas-rem) [inline tests/drivers/build_all/sensor/i2c.dtsi:1097]: You could add a value for `fs`, as the sensor supports dual full-scale.

lps28dfw defines

`fs = <LPS28DFW_DT_FS_MODE_1_1260>; `

File: drivers/sensor/st/lps2xdf/lps2xdf.c [hunk 2]
```diff
@@ -6,6 +6,7 @@
  * SPDX-License-Identifier: Apache-2.0
  *
  * Datasheet:
+ * https://www.st.com/resource/en/datasheet/ilps22qs.pdf
```
Comment (jonas-rem) [inline drivers/sensor/st/lps2xdf/lps2xdf.c:9]: You are adding a link to the datasheet in `lps2xdf.c `, but not in `lps2xdf.h´. 

What do you think of removing the set of link from one of the files?

File: drivers/sensor/st/lps2xdf/ilps22qs.c [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+static void ilps22qs_handle_interrupt(const struct device *dev)
+{
+}
+
+/**
+ * ilps22qs_trigger_set - not supported
+ */
+static int ilps22qs_trigger_set(const struct device *dev,
+			  const struct sensor_trigger *trig,
+			  sensor_trigger_handler_t handler)
```
Comment (jonas-rem) [inline drivers/sensor/st/lps2xdf/ilps22qs.c:71]: Indentation

**Review by jonas-rem — CHANGES_REQUESTED** [review]
Looks good to me, just saw some minor points.

Tested the updated driver with a custom board that has the `lps28dfw` device. I tested interrupt mode and both scales (1260 hPA and 4060 hPA). Works well.

Saw some unchanged copyright notices from 2023, not sure if you want to update them.

**Comment by avisconti** [comment]
@jonas-rem can you please test lps28dfw on your boards? Thanks

**Comment by jonas-rem** [comment]
> @jonas-rem can you please test lps28dfw on your boards? Thanks

@avisconti that's what I ment in my review comment with "custom board". I tested the lps28dfw with my board and everything works well.

**Comment by avisconti** [comment]
> > @jonas-rem can you please test lps28dfw on your boards? Thanks
> 
> @avisconti that's what I ment in my review comment with "custom board". I tested the lps28dfw with my board and everything works well.

Yes, sorry. Yesterday evening I was tired and I didn't see your feedback. Sorry. I'll review your inputs later on today. Thanks

File: tests/drivers/build_all/sensor/i2c.dtsi (hunk 1 above)
Comment (avisconti) [reply tests/drivers/build_all/sensor/i2c.dtsi:1097]: Yes, I have already defined ILPS22QS_DT_FS_MODE_1_1260, so I can put it here.

File: drivers/sensor/st/lps2xdf/lps2xdf.c (hunk 2 above)
Comment (avisconti) [reply drivers/sensor/st/lps2xdf/lps2xdf.c:9]: Yes, maybe in lps2xdf.h. 
Thanks for noticing.

File: drivers/sensor/st/lps2xdf/ilps22qs.c (hunk 3 above)
Comment (avisconti) [reply drivers/sensor/st/lps2xdf/ilps22qs.c]: ok, done

**Review by ubieda — DISMISSED** [review]
LGTM other than a couple typos.

File: dts/bindings/sensor/st,ilps22qs-spi.yaml [hunk 4]
```diff
@@ -0,0 +1,10 @@
+# Copyright (c) 2023 STMicroelectronics
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
+    STMicroelectronics LPS22DF pressure and temperature sensor connected to I2C
```
[the remark this answers is not in the document]
Comment (avisconti) [reply dts/bindings/sensor/st,ilps22qs-spi.yaml:5]: Thanks, I'll correct both of them

**Review by ubieda — DISMISSED** [review]

**Review by jonas-rem — APPROVED** [review]

**Comment by avisconti** [comment]
@ubieda unfortunately I had to rebase and push again to solve merge issues on tests/drivers/build_all/sensor/i2c.dtsi.

Any other would like to review this work? Thanks in advance

**Review by jonas-rem — APPROVED** [review]

**Review by ubieda — APPROVED** [review]

**Pull request merged**